### DISCUSSION TITLE
Post 23.nn Upgrade: Email field on user page is required?

### DISCUSSION THREAD

**Discussion by pawsey-kbuckley** [opendcim/openDCIM#1415] [Q&A] [answered]
This may be something specific to my instance but I thought I'd try and rule out
the behaviour I am seeing being a result of the upgrade to 23.02, as I don't recall
being ever bitten by it before.

Came to alter the rights for a user (none of our users have anything in the email
field), and, after clicking the Update button, I get a popup tellimg me that the email
field has an invalid value and that "This field is required".

I understand the need to have the validator run over any value in the email field but 
can't see why it would be REQUIRED, believing that that would be a site-local choice?

As I say though, maybe it's just my local instance that's screwed?

**Comment by pawsey-kbuckley** [comment]
Think I may have narrowed this one dowen a little, in that I went to another
23.02 instance and tried to replicate the issue.

If I take a user with no rights and try to add and remove rights one at a time,
I can seem to do so, all the way down the list until I get to the `Enter Rack Requests`
item.

Adding that right completes, but then trying to remove it generates the issue with the
email field.

Furthermore, once a user has had the `Enter Rack Requests` right assigned, attempts
to alter ANY other right then generate the issue.

As I can replicate my issue, I am going to create an Issue from this Discussiont, although
I appreciate that it still may be something specific to my deployments.

**Comment by pawsey-kbuckley** [comment]
I just thought to try things out at the openDCIM demo instance.

The ABC Corp user there has the "Enter Rack Requests" right set, and,
if I try to remove it I get the "This field is required" that I see on my instance

Not just me then: phew!

**Reply by samilliken** [reply, marked as answer]
The reason why it's required is that the Rack Requests system sends out an email, so if you don't have one, it won't work.   It's a designed requirement.

**Comment by pawsey-kbuckley** [comment]
Cheers for that Scott: I can now see why that would be an issue.

Just to clarify though: if a user doesn't have, or isn't being given,
either of these two rights

Enter Rack Requests
Complete Rack Requests

then there shouldn't be any designed requirement to have an email
address associated?

**Reply by samilliken** [reply]
I can't think of a reason to require it without those two.   However, given the persnickity nature of many JavaScript libraries, the error-checking on the fields might not be something that can be dynamically changed based upon other check boxes, so it may simply be enabled in all cases.   Either way, it's how it has been for pretty much a decade.

**Comment by wilpig** [comment]
https://github.com/opendcim/openDCIM/issues/1416
